pvlib/iotools/srml.py

```diff
@@ -134,11 +134,17 @@ def format_index(df):
     year = int(df.columns[1])
     df_doy = df[df.columns[0]]
     # Times are expressed as integers from 1-2400, we convert to 0-2359 by
-    # subracting one and then correcting the minutes at each former hour.
-    df_time = df[df.columns[1]] - 1
-    fifty_nines = df_time % 100 == 99
-    times = df_time.where(~fifty_nines, df_time - 40)
-
+    # subracting the length of one interval and then correcting the times
+    # at each former hour. interval_length is determined by taking the
+    # difference of the first two rows of the time column.
+    interval_length = int(df[df.columns[1]][:2].diff()[1])
```

Comment:
This line would be easier to read as, e.g., `interval_length = df[df.columns[0]][1] - df[df.columns[0]][0]`